MailCore/Cache/MailboxManager/MailboxManager.swift

```diff
@@ -180,26 +180,28 @@ public final class MailboxManager: ObservableObject, MailboxManageable {
     }
 
     func keepCacheAttributes(
-        for message: Message,
+        forLiveMessage liveMessage: Message,
         keepProperties: MessagePropertiesOptions,
         using realm: Realm
     ) {
-        guard let savedMessage = realm.object(ofType: Message.self, forPrimaryKey: message.uid) else {
+        guard let savedMessage = realm.object(ofType: Message.self, forPrimaryKey: liveMessage.uid) else {
             return
         }
-        message.inTrash = savedMessage.inTrash
+        liveMessage.inTrash = savedMessage.inTrash
         if keepProperties.contains(.fullyDownloaded) {
-            message.fullyDownloaded = savedMessage.fullyDownloaded
+            liveMessage.fullyDownloaded = savedMessage.fullyDownloaded
         }
         if keepProperties.contains(.body), let body = savedMessage.body {
-            message.body = Body(value: body)
+            liveMessage.body = Body(value: body).detached()
```

Comment:
No need to recreate from Body from value. We can create a new one with detached.
```suggestion
            liveMessage.body = body.detached()
```